src/basic/fileio.c

```diff
@@ -148,6 +148,28 @@ int write_string_stream_ts(
                         return -EBADF;
         }
 
+        if (flags & WRITE_STRING_VIRTUALFILE_SUPRESS_IF_POSSIBLE) {
+                _cleanup_free_ char *ret = NULL;
+
+                if (fd < 0) {
+                        fd = fileno(f);
+                        if (fd < 0)
+                                return -EBADF;
+                }
+
+                r = read_virtual_stream(fd, SIZE_MAX, &ret, NULL);
```

Comment:
the second argument can be set to strlen(line) or so, given that if the current contents is longer than what you want to write it can't be identical to what you intend to write